packages/react/src/utils/idRefs.ts

```diff
@@ -0,0 +1,41 @@
+type IdRefs = string | null | undefined;
+
+/**
+ * Returns a unique set of id refs from the provided string
+ * @param ids - string of id refs
+ */
+export default function idRefs(ids: IdRefs): Set<string> {
+  if (!ids || !ids.trim()) {
+    return new Set<string>();
+  }
+  return new Set(ids.trim().split(/\s+/));
+}
```

Comment:
Sets are un-ordered, so it's possible when adding or removing an ID that the order of the IDs change. Unless I'm missing something, we should change this implementation to make sure ordering is preserved, such as with an `Array`

Reply:
I think you're missing something. Sets maintain their order via [insertion order](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add):
> The insertion order corresponds to the order in which each element was inserted into the set by the [add()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add) method successfully (that is, there wasn't an identical element already in the set when add() was called).

Adding an id that already exists should be a noop as the id already exists, and it should always be appended to the end when it doesn't exist so there's no order change there. I'm not quite following on how removing an id should change the order of the set. The below test shows that the order still is the same after removal:

https://github.com/dequelabs/cauldron/blob/1a2038d03ab58ea716ae28df3280bf5cc800d160/packages/react/src/utils/idRefs.test.ts#L28-L30

Reply:
Cool! My Computer Science days about the definition of a Set were just haunting me, then. It's cool the JS maintains insertion order